kitsune/questions/api.py

```diff
@@ -22,60 +23,42 @@
 
 class QuestionMetaDataSerializer(serializers.ModelSerializer):
     question = serializers.PrimaryKeyRelatedField(
-        required=False, write_only=True)
+        required=False,
+        write_only=True,
+        queryset=Question.objects.all())
 
     class Meta:
         model = QuestionMetaData
         fields = ('name', 'value', 'question')
 
-    def get_identity(self, obj):
-        return obj['name']
-
-    def restore_object(self, attrs, instance=None):
-        """
-        Given a dictionary of deserialized field values, either update
-        an existing model instance, or create a new model instance.
-        """
-        if instance is not None:
-            for key in self.Meta.fields:
-                setattr(instance, key, attrs.get(key, getattr(instance, key)))
-            return instance
-        else:
-            obj, created = self.Meta.model.objects.get_or_create(
-                question=attrs['question'], name=attrs['name'],
-                defaults={'value': attrs['value']})
-            if not created:
-                obj.value = attrs['value']
-                obj.save()
-            return obj
-
 
 class QuestionTagSerializer(serializers.ModelSerializer):
-    question = serializers.PrimaryKeyRelatedField(
-        required=False, write_only=True)
-
     class Meta:
         model = Tag
         fields = ('name', 'slug')
 
 
 class QuestionSerializer(serializers.ModelSerializer):
+    answers = serializers.PrimaryKeyRelatedField(many=True, read_only=True)
     content = SplitSourceField(read_source='content_parsed', write_source='content')
     created = DateTimeUTCField(read_only=True)
-    creator = serializers.SerializerMethodField('get_creator')
-    involved = serializers.SerializerMethodField('get_involved_users')
-    is_solved = serializers.Field(source='is_solved')
-    is_taken = serializers.Field(source='is_taken')
-    metadata = QuestionMetaDataSerializer(source='metadata_set', required=False)
-    num_votes = serializers.Field(source='num_votes')
-    product = serializers.SlugRelatedField(required=True, slug_field='slug')
-    tags = QuestionTagSerializer(source='tags', read_only=True)
+    creator = serializers.SerializerMethodField()
```

Comment:
`SerializerMethodField` now have a sane default for the name of the method. Specifying a value that is equal to the sane default is an error. The default is `get_{variable_name}`.